### PR TITLE
Update piptools to fix `make requirements` error

### PR DESCRIPTION
**Pull request by rpenido**
## Description
Running `make requirements` outputs the following error:

```
pip install -qr requirements/pip-tools.txt
pip-sync requirements/dev.txt requirements/private.*
Traceback (most recent call last):
  File "/home/user/.virtualenvs/openedx-learning/bin/pip-sync", line 5, in <module>
    from piptools.scripts.sync import cli
  File "/home/user/.virtualenvs/openedx-learning/lib/python3.8/site-packages/piptools/scripts/sync.py", line 16, in <module>
    from .. import sync
  File "/home/user/.virtualenvs/openedx-learning/lib/python3.8/site-packages/piptools/sync.py", line 11, in <module>
    from pip._internal.commands.freeze import DEV_PKGS
ImportError: cannot import name 'DEV_PKGS' from 'pip._internal.commands.freeze' (/home/user/.virtualenvs/openedx-learning/lib/python3.8/site-packages/pip/_internal/commands/freeze.py)
make: *** [Makefile:61: requirements] Error 1
```
This seems to be related to an incompatibility. with pip 23.3

Updating piptools fixed the issue.

## More info
* https://github.com/jazzband/pip-tools/pull/1906

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @rpenido!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


This is currently a draft pull request. When it is ready for our review and all tests are green, click "Ready for Review", or remove "WIP" from the title, as appropriate.

File: requirements/base.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #
-# This file is autogenerated by pip-compile with Python 3.10
+# This file is autogenerated by pip-compile with Python 3.8
```
Comment (ChrisChV) [inline requirements/base.txt:2]: @rpenido It looks like you are running these commands with python 3.8. I tried `make requirements` on `main` and it works well on my computer. Maybe you need to run the command with python 3.10. Could you try that?

Comment (rpenido) [reply requirements/base.txt:2]: I have the same problem with python 3.10.13.

What's your pip version?
```
pip 23.2.1 from /home/user/.pyenv/versions/3.10.13/lib/python3.10/site-packages/pip (python 3.10)
```

It only works with pip 23.0.1 (python 3.8 or 3.10)

Comment (ChrisChV) [reply requirements/base.txt:2]: Ohh I see, my pip version is:

```
pip 23.1.2 from /media/xnpio/linuxDisk/xnpiochv/Documentos/edx/openedx-learning/venv/lib/python3.10/site-packages/pip (python 3.10)
```

I will try with the latest version

Comment (rpenido) [reply requirements/base.txt:2]: If you feel this is not an issue, feel free to close this PR!

I can downgrade and work with pip 23.0.1 for this project. I just wanted to know if someone was having the same problem.

I don't think we should spend too much time on this! :smiley:

Comment (ChrisChV) [reply requirements/base.txt:2]: Yep, is the version

Comment (ChrisChV) [reply requirements/base.txt:2]: > If you feel this is not an issue, feel free to close this PR!
> 
> I can downgrade and work with pip 23.0.1 for this project. I just wanted to know if someone was having the same problem.
> 
> I don't think we should spend too much time on this!

Yes I agree, I got some other errors using my python 3.10. I think we can leave as is now.

Comment (ChrisChV) [reply requirements/base.txt:2]: @rpenido I think they beat us to it a few days ago with https://github.com/openedx/openedx-learning/pull/83 :laughing:

Comment (rpenido) [reply requirements/base.txt:2]: Great! Thank you for your time here @ChrisChV!
Closing!

**Comment by openedx-webhooks** [comment]
@rpenido Even though your pull request wasn’t merged, please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/openedx-learning&entry.867055334=https://github.com/openedx/openedx-learning/pull/85&entry.1484655318=https://github.com/rpenido&entry.752974735=2023-09-26+13:12&entry.1917517419=2023-09-27+19:05&entry.2133058324=No) so we can improve your experience in the future.

**Pull request closed without merging**